config/v1/types_securityprofile.go

```diff
@@ -0,0 +1,304 @@
+package v1
+
+// SecurityProfile defines the schema for a security profile. This object is used
+// by operators to apply network security settings to operands.
```

Comment:
Is anyone besides Ingress going to use this?

Reply:
> @deads2k talked us into moving it out of ingress for other people to use... David, can you please identify who else you thought might use it? I know you had a decent argument, I just can't remember it.

@knewcomerRH wants openshift/sig-master to implement the ability for a cluster-admin to choose an TLS cipher suite and cipher versions.  I saw this API doing a targeted review and it it's a very good fit for our needs.  To avoid teaching admins about multiple different security profiles, we can use this inside of the apiservers.config.openshift.io to specify a security profile use the same core types as Ingress so that we only have to have one kind of security profile.

Reply:
@deads2k @knewcomerRH are you ok with one of the name changes that @smarterclayton suggests in https://github.com/openshift/api/pull/350#discussion_r305044110? Of the 3, I prefer `TLSSecurityProfile`.

Reply:
@danehans I also prefer TLSSecurityProfile.